include/git2/repository.h

```diff
@@ -746,6 +747,15 @@ GIT_EXTERN(const char *) git_repository_get_namespace(git_repository *repo);
 GIT_EXTERN(int) git_repository_is_shallow(git_repository *repo);
 
 /**
+ * Determine the shallow roots of the repository
+ *
+ * @param out An array of shallow oids. Can be NULL.
+ * @param repo The repository
+ * @return 1 if shallow, zero if not
```

Comment:
This behaves completely differently from the rest of the functions. This is not `git_reposistory_is_shallow`, this reads the file and should thud return 0 or an error code.